hazelcast/src/main/java/com/hazelcast/spi/impl/classicscheduler/ClassicOperationScheduler.java

```diff
@@ -0,0 +1,432 @@
+    private final OperationHandler[] partitionOperationHandlers;
+
+    //the generic workqueues are shared between all generic operation threads, so that work can be stolen
+    //and a task gets processed as quickly as possible.
+    private final BlockingQueue genericWorkQueue = new LinkedBlockingQueue();
+    private final ConcurrentLinkedQueue genericPriorityWorkQueue = new ConcurrentLinkedQueue();
+    //all operations that are not specific for a partition will be executed here, e.g heartbeat or map.size
+    private final GenericOperationThread[] genericOperationThreads;
+    private final OperationHandler[] genericOperationHandlers;
+
+    private final ResponseThread responseThread;
+    private final BasicResponsePacketHandler responsePacketHandler;
+    private final Address thisAddress;
+    private final NodeExtension nodeExtension;
+    private final HazelcastThreadGroup threadGroup;
+
+    //The trigger is used when a priority message is send and offered to the operation-thread priority queue.
+    //To wakeup the thread, a priorityTaskTrigger is send to the regular blocking queue to wake up the operation
+    //thread.
+    private final Runnable priorityTaskTrigger = new Runnable() {
+        @Override
+        public void run() {
+        }
+
+        @Override
+        public String toString() {
+            return "TriggerTask";
+        }
+    };
+
+    public ClassicOperationScheduler(GroupProperties properties,
+                                     LoggingService loggerService,
+                                     Address thisAddress,
+                                     OperationHandlerFactory operationHandlerFactory,
+                                     BasicResponsePacketHandler responsePacketHandler,
+                                     HazelcastThreadGroup hazelcastThreadGroup,
+                                     NodeExtension nodeExtension) {
+        this.thisAddress = thisAddress;
+        this.nodeExtension = nodeExtension;
+        this.threadGroup = hazelcastThreadGroup;
+        this.logger = loggerService.getLogger(ClassicOperationScheduler.class);
+        this.responsePacketHandler = responsePacketHandler;
+
+        this.partitionOperationHandlers = initPartitionOperationHandlers(properties, operationHandlerFactory);
+        this.partitionOperationThreads = initPartitionThreads(properties);
+
+        this.genericOperationHandlers = initGenericOperationHandlers(properties, operationHandlerFactory);
+        this.genericOperationThreads = initGenericThreads();
+
+        this.responseThread = initResponseThread();
+
+        logger.info("Starting with " + genericOperationThreads.length + " generic operation threads and "
+                + partitionOperationThreads.length + " partition operation threads.");
+    }
+
+    private OperationHandler[] initGenericOperationHandlers(GroupProperties properties, OperationHandlerFactory handlerFactory) {
```

Comment:
The init methods are in a different order than they are used, so I got a bit confused while reading this on Github (no code navigation/method highlighting). It would be very kind to re-order them.